This doesn't compile for me on OS X. Any clues?

```
/Users/nah/code/hydro_ws/src/strands_executive/scheduler/src/scipUser.cpp:343:9: error: no viable overloaded '='
  start = std::chrono::high_resolution_clock::now();
  ~~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../lib/c++/v1/chrono:750:29: note: candidate function (the implicit copy assignment operator) not
      viable: no known conversion from 'time_point<class std::__1::chrono::steady_clock, duration<[...], ratio<[...], 1000000000>>>' to 'const
      time_point<class std::__1::chrono::system_clock, duration<[...], ratio<[...], 1000000>>>' for 1st argument
class _LIBCPP_TYPE_VIS_ONLY time_point
                            ^
/Library/Developer/CommandLineTools/usr/bin/../lib/c++/v1/chrono:750:29: note: candidate function (the implicit move assignment operator) not
      viable: no known conversion from 'time_point<class std::__1::chrono::steady_clock, duration<[...], ratio<[...], 1000000000>>>' to
      'time_point<class std::__1::chrono::system_clock, duration<[...], ratio<[...], 1000000>>>' for 1st argument
class _LIBCPP_TYPE_VIS_ONLY time_point
                            ^
/Users/nah/code/hydro_ws/src/strands_executive/scheduler/src/scipUser.cpp:345:7: error: no viable overloaded '='
  end = std::chrono::high_resolution_clock::now();
  ~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../lib/c++/v1/chrono:750:29: note: candidate function (the implicit copy assignment operator) not
      viable: no known conversion from 'time_point<class std::__1::chrono::steady_clock, duration<[...], ratio<[...], 1000000000>>>' to 'const
      time_point<class std::__1::chrono::system_clock, duration<[...], ratio<[...], 1000000>>>' for 1st argument
class _LIBCPP_TYPE_VIS_ONLY time_point
                            ^
/Library/Developer/CommandLineTools/usr/bin/../lib/c++/v1/chrono:750:29: note: candidate function (the implicit move assignment operator) not
      viable: no known conversion from 'time_point<class std::__1::chrono::steady_clock, duration<[...], ratio<[...], 1000000000>>>' to
      'time_point<class std::__1::chrono::system_clock, duration<[...], ratio<[...], 1000000>>>' for 1st argument
class _LIBCPP_TYPE_VIS_ONLY time_point
                            ^
2 errors generated.

```

--- a/scheduler/CMakeLists.txt
+++ b/scheduler/CMakeLists.txt
@@ -113,7 +113,7 @@ include_directories(
 
 add_definitions(-std=c++0x -DNDEBUG -DWITH_SCIPDEF -DROUNDING_FE -DNPARASCIP -DWITH_ZLIB -DWITH_GMP -DWITH_READLINE -DWITH_ZIMPL -fomit-frame-pointer  -mtune=native -m64)
 
-add_library(scheduler src/task.cpp src/scheduler.cpp src/distWrapper.cpp src/scipUser.cpp)
+add_library(scheduler src/task.cpp src/scheduler.cpp src/distWrapper.cpp src/scipUser.cpp src/pairs.cpp)
 add_dependencies(scheduler scipoptsuite_ext)
 target_link_libraries(scheduler
   ${catkin_LIBRARIES}

--- a/scheduler/src/scheduler.cpp
+++ b/scheduler/src/scheduler.cpp
@@ -294,11 +294,13 @@ bool Scheduler::solve(int version,string filename)
   Pairs * pr = new Pairs(tasksToS);
   if(version==1) //Brian Coltin
     numPairs = pr->setPairs_BC();
-  if(version==2) //mine
+  else if(version==2) //mine
     numPairs = pr->setPairs_mine();
-  if(version==3) //robot version
+  else if(version==3) //robot version
     numPairs = pr->setPairs();
-  if(version==4) //mine specific approach
+  else if(version==4) //mine specific approach
+    numPairs = pr->setPairs_new();
+  else //if parameter is not set
     numPairs = pr->setPairs_new();
 
   pr->getPairs(&pairs);

--- a/scheduler/src/scheduler_node.cpp
+++ b/scheduler/src/scheduler_node.cpp
@@ -82,7 +82,7 @@ bool getSchedule(strands_executive_msgs::GetSchedule::Request  &req,
   messageStore.insert(spl);
 
   Scheduler scheduler(&tasks);
-  if(scheduler.solve()) {
+  if(scheduler.solve(version, filename)) {
   	std::sort(tasks.begin(), tasks.end(), compareTasks);
 
   	for(auto & tp : tasks) {

--- a/scheduler/src/scipUser.cpp
+++ b/scheduler/src/scipUser.cpp
@@ -336,7 +336,8 @@ SCIP_Retcode ScipUser::setFinalCons(vector<Task*> * tasksToS, vector<SCIP_VAR *>
 SCIP_Retcode ScipUser::scipSolve(vector<Task*> * tasksToS, SCIP_VAR * vars[], bool * worked, string filename)
 {
   int num_tasks = tasksToS -> size();
-  std::chrono::time_point<std::chrono::system_clock> start, end;
+  //std::chrono::time_point<std::chrono::system_clock> start, end;
+  std::chrono::high_resolution_clock::time_point start, end;
   ofstream results;
 
   SCIP_Real vals[num_tasks]; //array to save execution times
@@ -347,13 +348,19 @@ SCIP_Retcode ScipUser::scipSolve(vector<Task*> * tasksToS, SCIP_VAR * vars[], bo
  
   //SCIP_CALL( SCIPprintBestSol(scip, NULL, FALSE) );
   SCIP_SOL* sol = SCIPgetBestSol(scip);
-  results.open (filename,std::ios_base::app);
-  results << SCIPgetSolOrigObj(scip,sol) << " " << elapsed_seconds.count();
-  
+  if(!filename.empty())
+  {
+    results.open (filename,std::ios_base::app);
+    results << SCIPgetSolOrigObj(scip,sol) << " " << elapsed_seconds.count();
+  }
+
   if(sol == NULL)
   {
     *worked = false;
-    results << " " << 0 << "\n";
+    if(!filename.empty())
+    {
+      results << " " << 0 << "\n";
+    }
     for(int i=0; i < num_tasks; i++)
     {
       tasksToS->at(i)->setExecTime(-1.0);
@@ -362,7 +369,10 @@ SCIP_Retcode ScipUser::scipSolve(vector<Task*> * tasksToS, SCIP_VAR * vars[], bo
   else
   {
     *worked = true;
-    results << " " << 1 << "\n";
+    if(!filename.empty())
+    {
+      results << " " << 1 << "\n";
+    }
     SCIP_CALL(SCIPgetSolVals(scip,sol, num_tasks, vars, vals)); 
     for(int i=0; i < num_tasks; i++)
     {

